tensorboard/plugins/image/summary.py

```diff
@@ -129,7 +131,35 @@ def pb(name, images, max_outputs=3, display_name=None, description=None):
   return summary
 
 
+class _TensorFlowPNGEncoder(object):
+
+  def __init__(self):
+    super(_TensorFlowPNGEncoder, self).__init__()
+    self._session = None
+    self._image_placeholder = None
+    self._encode_op = None
+    self._initialization_lock = threading.Lock()
+
+  def _lazily_initialize(self):
+    with self._initialization_lock:
```

Comment:
I'd recommend the double-checked locking pattern here, but there wouldn't be any benefit to doing that in Python, and probably not possible without volatile. Look that pattern up if you're not familiar with it.